add release note for 0.5 (#69)
apache/celeborn-website: docs/community/release_notes/release_note_0.5.0.md

---
hide:
  - navigation

license: |
  Licensed to the Apache Software Foundation (ASF) under one or more
  contributor license agreements.  See the NOTICE file distributed with
  this work for additional information regarding copyright ownership.
  The ASF licenses this file to You under the Apache License, Version 2.0
  (the "License"); you may not use this file except in compliance with
  the License.  You may obtain a copy of the License at

      https://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
---

# Apache Celeborn™ 0.5.0 Release Notes

## Highlight

> - Celeborn support TLS for communication

Reviewer: ```suggestion
- Support TLS for internal communication
```

- Memory usage and performance optimization
- Enhanced metrics and HTTP APIs

> - Support Spark3.5 with JDK21

Reviewer: ```suggestion
- Support Spark 3.5 with JDK 21
```

Reviewer: actually, this is a little bit contrived, Spark 3.5 does not officially support Java 21

Reviewer: Maybe just "Support Java 21"?

- [CELEBORN-1134][FOLLOWUP] Add execution.batch-shuffle-mode: ALL_EXCHANGES_BLOCKING to Flink Configuration of Deploy Flink client
- [MINOR] Fix typo of celeborn.network.bind.preferIpAddress doc
- [MINOR] Fix typo in developer docs - overview
- [MINOR] Fix style and Gluten link in Developers Doc

### Dependencies
- [CELEBORN-1400] Bump Ratis version from 2.5.1 to 3.0.1
- [CELEBORN-1396] Bump Netty from 4.1.107.Final to 4.1.109.Final
- [CELEBORN-1395] Bump RoaringBitmap version from 1.0.5 to 1.0.6
- [CELEBORN-1394] Bump Spark from 3.4.2 to 3.4.3
- [CELEBORN-1389] Bump Dropwizard version from 3.2.6 to 4.2.25
- [CELEBORN-1382] Bump RoaringBitmap version from 0.9.32 to 1.0.5
- [CELEBORN-1366] Bump guava from 32.1.3-jre to 33.1.0-jre
- [CELEBORN-1330] Bump rocksdbjni version from 8.5.3 to 8.11.3
- [CELEBORN-1281] Bump Spark from 3.5.0 to 3.5.1
- [CELEBORN-1262] Bump Spark from 3.3.3 to 3.3.4
- [CELEBORN-1243] Bump Flink from 1.18.0 to 1.18.1
- [CELEBORN-1221] Bump Flink from 1.17.0 to 1.17.2
- [MINOR] Bump minikube and kubernetes version of integration test
- [BUILD] Bump netty version to latest 4.1.107.Final

## Credits

> Thanks to the following contributors who helped to review and commit to Apache Celeborn 0.4.1 version:

Reviewer 2: ```suggestion
Thanks to the following contributors who helped to review and commit to Apache Celeborn 0.5.0 version:
```

| Contributors |                 |                   |                 |                     |               |
|--------------|-----------------|-------------------|-----------------|---------------------|---------------|
| albin3       | AngersZhuuuu    | Aravind Patnam    | binjie yang     | cfmcgrady           | ChenYi015     |
| CodingCat    | Curtis Howard   | cxzl25            | dev-lpq         | ErikFang            | FMX           |
| ForVic       | huangxiaopingRD | jiang13021        | jiaoqingbo      | kerwin-zk           | labbomb       |
| leixm        | lifulong        | lyy-pineapple     | miyuesc         | Mridul Muralidharan | mridulm       |
| onebox-li    | otterc          | pan3793           | radeity         | s0nskar             | SteNicholas   |

> | tiny-dust    |   turboFei      |      waitinfuture |       wxplovecc |   xianminglei       |    xinyuwang1 |

Reviewer: lint

Author: Fixed
